src/engine/pieces/queen.ts

```diff
@@ -1,13 +1,43 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from "../square";
+import gameSettings from "../gameSettings";
 
 export default class Queen extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        const currentSquare = board.findPiece(this)
```

Comment:
The queen's possible moves are the combination of a rook and bishop. Do you think you can commonise some logic between them? 🙂